dvc/__init__.py

```diff
@@ -96,3 +96,8 @@ def _remove_version_file():
     warnings.simplefilter("ignore", CryptographyDeprecationWarning)
 except ImportError:
     pass
+
+from dvc.scm import SCM  # noqa: E402
+from dvc.git.scm import Git  # noqa: E402
+
+SCM.register(Git)
```

Comment:
@ei-grad these 3 lines look like a hack, or am I missing something? Why just moving tree outside is not enough to break the cycle?

Reply:
My point is the following:

We can and it would be enough to fix the tests. But the actual problem here is that the `Git` import in `dvc/scm/__init__.py` is the wrong thing itself - it denotes that `dvc.scm` depends on `Git`, and this introduces the cycle dependency because actually the `Git` depends on `scm.Base` (`Git` inherits `scm.Base`) and `Base` is the part of `scm`. Python could manage this cycle dependency since `Git` imports the `Base` not like as the part of `scm`, but as the `scm.base`, but this the _real_ hack.

It could look like the right thing if `Git` would have been a part of `scm`, but it is not - it is more than a part, it is the interface implementation. Base is the interface and SCM is the method to select the implementation to access through this interface.

The right way to deal with this situation is to decouple SCM from Git, because the SCM doesn't really needs to know about the implementations of Base.

But maybe I just got stoned :-).